server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -1007,6 +1008,67 @@ def group_by_lines(lines: Sequence[int], column: np.ndarray) -> list[np.ndarray]
     return full_line_groups
 
 
+@dataclasses.dataclass(frozen=True, slots=True)
+class JIRAGroup:
+    """Grouping definition based on jira entities.
+
+    `None` field values indicate that no grouping should be done on the corresponding dataframe
+    property.
+    On the opposite an single empty field value makes this group select nothing.
+    """
+
+    projects: Optional[Collection[str]]
+    priorities: Optional[Collection[str]]
+    types: Optional[Collection[str]]
+
+    def __bool__(self) -> bool:
+        """Get whether the group definition is empty.  An empty group does not do any grouping."""
+        return any(f is not None for f in (self.projects, self.priorities, self.types))
+
+    @classmethod
+    def total(cls) -> JIRAGroup:
+        """Return a `JIRAGroup` selecting all data."""
+        return cls(None, None, None)
```

Comment:
btw, can we do the same thing by adding default values in the dataclass?
```python
class JIRAGrouping:
    projects: Optional[...] = None
```
I don't have a preference, just found out it was possible in the last PR.